From 2fe454f21b6b3890b02f52810a662530ec019968 Mon Sep 17 00:00:00 2001
From: mulhern <amulhern@redhat.com>
Date: Wed, 19 Aug 2020 11:11:14 -0400
Subject: [PATCH] Remove stratis module

It is obsolete. An approach which does not start the daemon is being worked
on.

Signed-off-by: mulhern <amulhern@redhat.com>
(cherry picked from commit fe761330e5db60f19f0ab2d937ea71a5b2438656)

Cherry-picked from: fe761330e5db60f19f0ab2d937ea71a5b2438656
Resolves: #1950572
---
 dracut.spec                               |  1 -
 modules.d/90stratis/module-setup.sh       | 33 -------------------------------
 modules.d/90stratis/stratisd-init.service | 15 --------------
 modules.d/90stratis/stratisd-start.sh     |  3 ---
 modules.d/90stratis/stratisd-stop.sh      |  6 ------
 5 files changed, 58 deletions(-)

diff --git a/dracut.spec b/dracut.spec
index fa3e4b99..dcaafa61 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -365,7 +365,6 @@ install -m 0755 51-dracut-rescue-postinst.sh $RPM_BUILD_ROOT%{_sysconfdir}/kerne
 %{dracutlibdir}/modules.d/90lvm
 %{dracutlibdir}/modules.d/90mdraid
 %{dracutlibdir}/modules.d/90multipath
-%{dracutlibdir}/modules.d/90stratis
 %{dracutlibdir}/modules.d/90qemu
 %{dracutlibdir}/modules.d/91crypt-gpg
 %{dracutlibdir}/modules.d/91crypt-loop
diff --git a/modules.d/90stratis/module-setup.sh b/modules.d/90stratis/module-setup.sh
deleted file mode 100755
index 943f572e..00000000
--- a/modules.d/90stratis/module-setup.sh
+++ /dev/null
@@ -1,33 +0,0 @@
-#!/bin/bash
-
-# called by dracut
-check() {
-    require_binaries stratisd-init thin_check thin_repair mkfs.xfs xfs_admin xfs_growfs || return 1
-    return 255
-}
-
-# called by dracut
-depends() {
-    echo dm
-    return 0
-}
-
-# called by dracut
-installkernel() {
-    instmods xfs
-}
-
-# called by dracut
-install() {
-
-    inst_multiple stratisd-init thin_check thin_repair mkfs.xfs xfs_admin xfs_growfs
-
-    if dracut_module_included "systemd"; then
-        inst_simple "${moddir}/stratisd-init.service" "${systemdsystemunitdir}/stratisd-init.service"
-        systemctl -q --root "$initdir" enable stratisd-init.service
-    else
-        inst_hook cmdline 25 "$moddir/stratisd-start.sh"
-        inst_hook cleanup 25 "$moddir/stratisd-stop.sh"
-    fi
-}
-
diff --git a/modules.d/90stratis/stratisd-init.service b/modules.d/90stratis/stratisd-init.service
deleted file mode 100644
index 318e8c27..00000000
--- a/modules.d/90stratis/stratisd-init.service
+++ /dev/null
@@ -1,15 +0,0 @@
-[Unit]
-Description=A daemon that manages a pool of block devices to create flexible file systems
-Documentation=man:stratisd(8)
-Before=local-fs-pre.target
-DefaultDependencies=no
-
-[Service]
-Type=simple
-ExecStart=/sbin/stratisd-init --debug
-KillSignal=SIGINT
-StandardOutput=syslog
-StandardError=syslog
-
-[Install]
-WantedBy=sysinit.target
diff --git a/modules.d/90stratis/stratisd-start.sh b/modules.d/90stratis/stratisd-start.sh
deleted file mode 100755
index afcd81fd..00000000
--- a/modules.d/90stratis/stratisd-start.sh
+++ /dev/null
@@ -1,3 +0,0 @@
-#!/bin/sh
-
-stratisd-init --debug > /dev/kmsg 2>&1 &
\ No newline at end of file
diff --git a/modules.d/90stratis/stratisd-stop.sh b/modules.d/90stratis/stratisd-stop.sh
deleted file mode 100755
index f394a843..00000000
--- a/modules.d/90stratis/stratisd-stop.sh
+++ /dev/null
@@ -1,6 +0,0 @@
-#!/bin/sh
-
-[ -f /lib/dracut-lib.sh ] && . /lib/dracut-lib.sh
-
-pid=$(pidof stratisd-init)
-[ -n "$pid" ] && kill ${pid}

